### PR TITLE
drivers: usb_dc_sam: free endpoint memory on End of Reset event 

### LINKED ISSUE #24626: USB re-connection fails on SAM E70
**Describe the bug**
On the SAM E70 SoC, the first USB device connection works, but all subsequent re-connections fail (host fails to detect the device and "failed to configure endpoint" is logged).

**To Reproduce**
1. Launch `samples/subsys/usb/cdc_acm` or any other USB device sample with the device USB cable connected.
2. Un-plug the device USB cable and then re-plug.

**Expected behavior**
Host successfully detects the device upon re-connection.

**Screenshots or console output**
Running `samples/subsys/usb/cdc_acm` on `sam_e70_xplained`
```
*** Booting Zephyr OS build zephyr-v2.2.0-1811-g799e725bec40  ***
[00:00:00.006,000] <inf> usb_cdc_acm: USB device suspended
[00:00:00.006,000] <inf> cdc_acm_echo: Wait for DTR
[00:00:00.009,000] <inf> usb_cdc_acm: USB device suspended
[00:00:00.384,000] <inf> usb_cdc_acm: USB device configured
[00:00:11.017,000] <inf> cdc_acm_echo: DTR set
[00:00:12.025,000] <inf> cdc_acm_echo: Baudrate detected: 115200
>>DISCONNECT USB CABLE<<
[00:00:15.206,000] <inf> usb_cdc_acm: USB device suspended
>>RECONNECT USB CABLE<<
[00:00:16.662,000] <err> usb_device: Failed to configure endpoint 0x81
[00:00:16.662,000] <err> usb_device: Failed to configure endpoint 0x83
[00:00:16.662,000] <err> usb_device: Failed to configure endpoint 0x02
[00:00:16.662,000] <inf> usb_cdc_acm: USB device configured
[00:00:16.850,000] <err> usb_device: Failed to configure endpoint 0x81
[00:00:16.850,000] <err> usb_device: Failed to configure endpoint 0x83
[00:00:16.850,000] <err> usb_device: Failed to configure endpoint 0x02
[00:00:16.850,000] <inf> usb_cdc_acm: USB device configured
[00:00:17.039,000] <err> usb_device: Failed to configure endpoint 0x81
[00:00:17.039,000] <err> usb_device: Failed to configure endpoint 0x83
[00:00:17.039,000] <err> usb_device: Failed to configure endpoint 0x02
[00:00:17.039,000] <inf> usb_cdc_acm: USB device configured
[00:00:17.228,000] <inf> usb_cdc_acm: USB device suspended
```

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Toolchain: Zephyr SDK 0.11.2
 - Commit SHA: bd639fc332675278132c57f18a196e23f0287115

**Additional context**
* Tested on `sam_e70_xplained` with the Rev. B chip
* The same issue is expected on `sam_v71_xplained` (cc @nandojve)

### PR DESCRIPTION
**Pull request by jfischer-no**
Free endpoint memory on End of Reset event (EORST).
Fixes: #24626

I also tweaked logging a little to see what happens during configuration, but I can remove commit f689a5a if not desired.

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
LGTM

**Review by aurel32 — APPROVED** [review]
According to the datasheet, this should not be needed as the endpoint is supposed to be disabled and its memory de-allocated upon receiving a USB reset. However I confirm this is not what happens on the hardware and that your changes actually fix the issue. Thanks!

File: drivers/usb/device/usb_dc_sam.c [hunk 1]
```diff
@@ -469,7 +469,7 @@ int usb_dc_ep_configure(const struct usb_dc_ep_cfg_data *const cfg)
 		return -EBUSY;
 	}
 
-	LOG_DBG("ep %x, mps %d, type %d", cfg->ep_addr, cfg->ep_mps,
+	LOG_INF("Configure ep %x, mps %d, type %d", cfg->ep_addr, cfg->ep_mps,
```
Comment (nandojve) [inline drivers/usb/device/usb_dc_sam.c:478]: On my view we can keep logs as LOG_DBG.

**Review by nandojve — APPROVED** [review]
@jfischer-phytec-iot thank you for the PR.

**Review by stephanosio — APPROVED** [review]
Although the reason why EP 1-3 do not get disabled after reset, as stated in the datasheet, may require further investigation, this patch does fix the problem described in #24626.

```
[00:00:09.270,000] <inf> usb_cdc_acm: USB device configured
>>from EORST ISR<<
EP 0: enabled
EP 1: enabled
EP 2: enabled
EP 3: enabled
EP 4: disabled
EP 5: disabled
EP 6: disabled
EP 7: disabled
EP 8: disabled
EP 9: disabled
```

**Pull request merged**